src/main/java/mekanism/common/util/MekanismUtils.java

```diff
@@ -777,8 +777,12 @@ public static void veinMineArea(IEnergyContainer energyContainer, Level world, B
                 continue;
             }
             BlockState targetState = world.getBlockState(foundPos);
-            FloatingLong destroyEnergy = destroyEnergyFunction.apply(targetState.getDestroySpeed(world, foundPos))
-                  .multiply(distanceMultiplier.applyAsDouble(WorldUtils.distanceBetween(pos, foundPos)));
+            float hardness = targetState.getDestroySpeed(world, foundPos);
+            if (targetState.isAir() || hardness == -1) {
```

Comment:
We should check and skip on air before bothering to check what the destroy speed is in case any mods that add a custom air block behave strangely/have expensive destroy speed logic.